lib/cluster/membership.c

```diff
@@ -1216,11 +1214,7 @@ known_node_cache_refresh_helper(xmlNode *xml_node, void *user_data)
         g_hash_table_replace(known_node_cache, uniqueid, node);
 
     } else if (pcmk_is_set(node->flags, crm_node_dirty)) {
-        if (!pcmk__str_eq(uname, node->uname, pcmk__str_casei)) {
-            free(node->uname);
-            node->uname = strdup(uname);
-            CRM_ASSERT(node->uname != NULL);
-        }
+        pcmk__copy_or_null(&node->uname, uname);
```

Comment:
a little more picky deciding if it needs to copy than it would have to be with a uname maybe but shouldn't hurt ...

Reply:
Yep, at worst, it unnecessarily frees and reallocates the name with a different case, but everything works the same. I did consider taking an argument for string flags e.g. pcmk__str_casei but I didn't see enough cases like this to be worth it.